src/connection/host_lookup.c

```diff
@@ -0,0 +1,444 @@
+
+  strcpy (hp.h_name, hostent_List[find_index].hostname);
+  strcpy (hp.h_aliases[0], hostent_List[find_index].hostname);
+  memcpy (hp.h_addr_list[0], addr_transform_buf2, sizeof (hp.h_addr_list[0]));
+
+  return &hp;
+}
+
+static int
+host_conf_load ()		//set hash table to discover error
+{
+  FILE *fp;
+  char file_line[HOSTNAME_BUF_SIZE + 1];
+  char line_buf[HOSTNAME_BUF_SIZE + 1];
+  char host_conf_file_full_path[PATH_MAX];
+  char *hosts_conf_dir;
+
+  char *token;
+  char *save_ptr_strtok;
+  /*delimiter */
+  char *delim = " \t\n";
+
+  int host_count = 0;
+
+  /*False, if hostent_List[index].hostname is set */
+  bool hostent_flag = INSERT_IPADDR;
+
+  memset (file_line, 0, HOSTNAME_BUF_SIZE + 1);
+  memset (line_buf, 0, HOSTNAME_BUF_SIZE + 1);
+
+  hosts_conf_dir = envvar_confdir_file (host_conf_file_full_path, PATH_MAX, "hosts.conf");
+  fp = fopen (hosts_conf_dir, "r");
+  if (fp == NULL)
+    {
+//err_msg : file open fail
+      return -1;
+      //error msg-> temp
+    }
+
+  while (fgets (file_line, HOSTNAME_BUF_SIZE + 1, fp) != NULL)
+    {
+//verifying duplicated hostname will be implemented soon
+      if (file_line[0] == '#')
+	continue;
+      if (file_line[0] == '\n')
+	continue;
+
+      token = strtok_r (file_line, delim, &save_ptr_strtok);
+      hostent_flag = INSERT_IPADDR;
+
+      char map_ipaddr[IPADDR_LEN];
+      char map_hostname[HOSTNAME_BUF_SIZE + 1];
+
+      do
+	{
+	  if (*token == '\0' || *token == '#')
+	    {
+	      break;
+	    }
+	  if (!hostent_flag)
+	    {
+	      if (sizeof (token) > HOSTNAME_BUF_SIZE + 1)
+		{
+//err_msg 
+		  return NULL;
+		}
+	      strcpy (hostent_List[host_count].ipaddr, token);
+	      strcpy (map_ipaddr, token);
+
+	      hostent_flag = INSERT_HOSTNAME;
+	    }
+	  else
+	    {
+	      if (sizeof (token) > IPADDR_LEN)
+		{
+//err_msg 
+		  return NULL;
+		}
+
+	      strcpy (hostent_List[host_count].hostname, token);
+	      strcpy (map_hostname, token);
+	    }
+	}
+      while (token = strtok_r (NULL, delim, &save_ptr_strtok));
+//naming rule check, detailed rule, error set
+//if hostent_List[host_count].hostname is empty, set error
+      if (hostent_List[host_count].hostname != NULL)
+	{
+	  if ((user_host_Map.find (map_hostname) == user_host_Map.end ()))
+	    {
+	      user_host_Map[map_hostname] = map_ipaddr;
+	    }
+/*duplicated hostname*/
+	  else if (!strcmp (user_host_Map.find (map_hostname)->second.c_str (), hostent_List[host_count].ipaddr))
+	    {
+/*duplicated hostname and ip address both*/
+	    }
+	  else
+	    {
+/*duplicated hostname but different ip address*/
+//err_msg : duplicated 
+	      return NULL;
+	    }
+	  host_count++;
+	}
+    }
+  return host_count;
+}
+
+/*
+ * gethostbyname_uhost () - Do same job with gethostbyname (), using by the 'user' defined 'hosts.conf' file.
+ * 
+ * return   : the hostent pointer.
+ * hostname (in) : the hostname.
+ *
+ */
+struct hostent *
+gethostbyname_uhost (char *name)
+{
+  struct hostent *hp;
```

Comment:
If you accept my suggestion from the line below, remove this line.

Reply:
In my opinion, this code would break the consistency.

In the gethostbyname_r, there is the code like below. 

```
  else
    {
      hp_buf = host_lookup_internal (name, NULL, HOSTNAME_TO_IPADDR);
      memcpy ((void *) ret, (void *) hp_buf, sizeof (struct hostent));
    }
```

So, it would decrease the readability.

Reply:
I applied it.